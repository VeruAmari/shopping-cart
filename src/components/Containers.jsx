import styled from 'styled-components';

const DefaultContainer = styled.div``;

export { DefaultContainer };

// 27 abril 221
