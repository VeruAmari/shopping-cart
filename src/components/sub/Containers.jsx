import styled from 'styled-components';

const PageContainer = styled.div`
  background-color: var(--color-5);
  display: grid;
  align-items: center;
  height: 100vh;
  width: 100vw;
  box-sizing: content-box;
`;

const AsideContainer = styled.div`
  background-color: var(--color-2);
  color: var(--color-5);
`;

export { PageContainer, AsideContainer };

// 27 abril 221
