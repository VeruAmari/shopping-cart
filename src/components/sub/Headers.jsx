import styled from 'styled-components';
import PropTypes from 'prop-types';

const Heading = ({ level, children, margin }) => {
  const cleanMargin = +margin;
  const tag = `h${level}`;
  return (
    <Wrapper
      style={{
        margin: `${!(cleanMargin === undefined) && !isNaN(cleanMargin) && !(cleanMargin === null) ? cleanMargin : '0.5'}rem`,
      }}
      as={tag}
    >
      {children}
    </Wrapper>
  );
};

const Wrapper = styled.h1`
  display: flex;
  justify-content: center;
`;

Heading.defaultProps = {
  level: 1,
};

Heading.propTypes = {
  level: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  margin: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.arrayOf(PropTypes.element),
  ]),
};

export default Heading;
