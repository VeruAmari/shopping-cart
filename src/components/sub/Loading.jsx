import styled from 'styled-components';

const Loading = () => {
  return (
    <Wrapper>
      <SpinningCircle />
      Loading...
    </Wrapper>
  );
};

const SpinningCircle = styled.div`
  box-sizing: border-box;
  border-left: 10px solid var(--color-3, pink);
  border-right: 10px solid var(--color-3, pink);
  border-top: 10px solid transparent;
  border-bottom: 10px solid transparent;
  border-radius: 100%;
  height: 100px;
  width: 100px;

  animation: 2s rotate infinite alternate ease-in-out;

  @keyframes rotate {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
`;

const Wrapper = styled.div`
  color: var(--color-3);
  display: grid;
  justify-items: center;
  font-weight: 900;
`;

export default Loading;
