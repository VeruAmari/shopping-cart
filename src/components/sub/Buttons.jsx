import styled from 'styled-components';
import { AsideContainer } from './Containers';

const LightColorfulButton = styled.button`
  background-color: var(--color-4);
  color: var(--color-2);
  border: 1px solid var(--color-2);
  &:hover {
    background-color: var(--color-2);
    color: var(--color-5);
    border-color: var(--color-2);
  }
  ${AsideContainer} & {
    border-color: var(--color-4);
  }
  ${AsideContainer} &:hover {
    background-color: var(--color-5);
    color: var(--color-1);
    border-color: var(--color-1);
  }
`;

const DarkButton = styled.button`
  background-color: var(--color-1);
  color: var(--color-4);
  border: 1px solid var(--color-1);
  &:hover {
    background-color: var(--color-4);
    color: var(--color-1);
    border-color: var(--color-1);
  }
`;

export { LightColorfulButton, DarkButton };
