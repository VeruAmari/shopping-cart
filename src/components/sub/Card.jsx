import styled from 'styled-components';
import PropTypes from 'prop-types';
import Heading from './Headers';
import { AsideContainer } from './Containers';

const radius = '7px';

const Card = ({ title, id, price, image, cartProducts, cb }) => {
  const onClickIncrease = (id, callBack) => {
    callBack((current) => {
      const newValue = current[id] ? current[id] + 1 : 1;

      return { ...current, [id]: newValue };
    });
  };

  const onClickDecrease = (id, callBack) => {
    callBack((current) => {
      const newValue = current[id] ? current[id] - 1 : 0;

      return { ...current, [id]: newValue };
    });
  };

  const isOnCart = !!cartProducts[id];

  const conditionalStyle = {
    outline: `${isOnCart ? '3px var(--color-1) solid' : 'none'}`,
  };

  return (
    <Wrapper style={conditionalStyle}>
      <ImageWrapper>
        <StyledImg src={image} alt={title} />
      </ImageWrapper>
      <FixedAsideContainer />
      <ProductName>{title}</ProductName>
      <ContentWrapper>
        <Heading level="3">${price}</Heading>
        <ButtonsContainer>
          <MinusButton
            onClick={() => {
              onClickDecrease(id, cb);
            }}
          >
            -
          </MinusButton>
          <CurrentAmount
            as="input"
            type="tel"
            onChange={(event) => {
              cb((current) => {
                const value = Number(event.target.value);
                const newValue = Number.isNaN(value) ? 0 : value;
                return {
                  ...current,
                  [id]: newValue,
                };
              });
            }}
            value={cartProducts[id] ? cartProducts[id] : '0'}
          />
          <PlusButton
            onClick={() => {
              onClickIncrease(id, cb);
            }}
          >
            +
          </PlusButton>
        </ButtonsContainer>

        <Heading level="4"></Heading>
      </ContentWrapper>
    </Wrapper>
  );
};

const FixedAsideContainer = styled(AsideContainer)`
  height: 0.5rem;
`;

const ProductName = styled.p`
  display: grid;
  justify-content: center;
  padding-top: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
`;

const SharedStyles = styled.div`
  font-size: 1rem;
  justify-content: center;
  align-content: center;
  text-align: center;
  width: 2rem;
  background-color: var(--color-5);
  color: var(--color-1);
  height: 2rem;
  display: grid;
  padding: 0;
`;

const CurrentAmount = styled(SharedStyles)`
  padding-left: 0.1rem;
  padding-right: 0.1rem;
  border: none;
`;

const SharedButtonStyles = styled(SharedStyles)`
  user-select: none;
  font-size: 2rem;
  border-radius: 5px;
  padding-left: 0.2rem;
  padding-right: 0.2rem;
  &:hover {
    color: var(--color-5);
  }
`;

const PlusButton = styled(SharedButtonStyles)`
  border-top-left-radius: 0%;
  border-left: 1px dotted var(--color-1);
  border-bottom-left-radius: 0%;
  &:hover {
    background-color: var(--color-3);
  }
`;

const MinusButton = styled(SharedButtonStyles)`
  border-right: 1px dotted var(--color-1);
  border-top-right-radius: 0%;
  border-bottom-right-radius: 0%;
  &:hover {
    background-color: var(--color-2);
  }
`;

const ButtonsContainer = styled.div`
  align-self: end;
  justify-items: center;
  justify-content: center;
  width: 100%;
  display: grid;
  grid-template-columns: auto auto auto;
`;

const Wrapper = styled.div`
  box-shadow: 5px 5px 15px rgba(0, 0, 0, 0.2);
  grid-template-rows: auto 1fr;
  align-items: space-between;
  display: grid;
  width: 250px;
  height: auto;
  max-height: 500px;
  border-radius: ${radius};
  background: radial-gradient(var(--color-5), var(--color-3));
  &:hover {
    box-shadow: 5px 5px 15px rgba(20, 0, 0, 0.5);
  }
`;

const ImageWrapper = styled.div`
  border-top-left-radius: ${radius};
  border-top-right-radius: ${radius};
  display: grid;
  justify-content: center;
  justify-items: center;
  background-color: white;
  width: 100%;
  max-height: 200px;
`;

const ContentWrapper = styled.div`
  align-items: start;
  align-content: end;
  padding: 0.5rem;
  display: grid;
  justify-content: center;
  justify-items: center;
  overflow: scroll;
  max-height: 100%;
  @media (max-width: 600px) {
    font-size: 1rem;
  }
`;

const StyledImg = styled.img`
  border-top-left-radius: ${radius};
  border-top-right-radius: ${radius};
  max-width: 100%;
  max-height: 200px;
`;

export const CardsContainer = styled.div`
  padding: 1rem;
  justify-items: center;
  grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
  grid-template-rows: auto;
  grid-gap: 1rem;
  display: grid;
`;

Card.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  price: PropTypes.number,
  category: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  cb: PropTypes.func,
  cartProducts: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

export default Card;
