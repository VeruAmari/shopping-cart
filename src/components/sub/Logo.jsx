import styled from 'styled-components';
import { NavElement } from './Containers';
import { HeaderWrapper } from '../main/Header';
import PropTypes from 'prop-types';

const Logo = ({ size }) => {
  return <LogoWrapper size={size}>Scui</LogoWrapper>;
};

const LogoWrapper = styled.span`
  font-family: var(--logo-font-family);
  font-style: var(--logo-font-style);
  font-weight: var(--logo-font-weight);

  font-size: ${(props) => props.size * 1.2 + 'rem'};

  ${HeaderWrapper} > & {
    color: var(--color-5);
  }
  ${NavElement} > & {
    color: var(--color-5);
  }
  ${NavElement}:hover > & {
    color: var(--color-1);
  }
`;

Logo.defaultProps = {
  size: '1',
};

Logo.propTypes = {
  size: PropTypes.string,
};

export default Logo;
