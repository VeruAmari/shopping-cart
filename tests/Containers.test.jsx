import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

// import userEvent from "@testing-library/user-event";

import { NavElement } from '../src/components/sub/Containers';

describe('NavElement', () => {
  it('renders', () => {
    render(
      <MemoryRouter>
        <NavElement>Cart</NavElement>
      </MemoryRouter>,
    );

    expect(screen.getByText('Cart')).toBeInTheDocument();
  });
});
