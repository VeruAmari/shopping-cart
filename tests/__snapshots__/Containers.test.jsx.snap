// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavElement > matches snapshot 1`] = `
<a
  class="Containers__NavElement-sc-1lz7g2b-1 fyMQmt"
  href="/"
>
  Cart
</a>
`;
